Implement client

IhorBerehovskyi opened the pull request:
Look at commits messages

IhorBerehovskyi committed:
Implemented base fuctional

IhorBerehovskyi committed:
Implement client menu and basic chart

IhorBerehovskyi committed:
Imlemented outdoor measuring chart line

IhorBerehovskyi committed:
Added settings fragment

IhorBerehovskyi committed:
General client refactoring

+Network api placed in a separate folder
+Deleted unnecessary file
+Fragments placed in a separate folder

IhorBerehovskyi committed:
Implemented singleton pattern for network connection

IhorBerehovskyi committed:
Merged with server update

IhorBerehovskyi committed:
Implemented mDNS, client requests for LED state

Sage-Cat commented on server/src/main.cpp:
```diff
@@ -94,6 +91,19 @@ int main(int argc, char **argv)
+    }
+    status = std::system("sudo systemctl restart avahi-daemon");
+    if (status != 0) {
+        std::cerr << "Error restarting avahi-daemon. Status: " << status
```
Use SPDLOG (for my code also, I just forgot it)

std::cout we need only for `printUsage`

IhorBerehovskyi replied:
Done

Sage-Cat commented on server/src/main.cpp:
```diff
@@ -94,6 +91,19 @@ int main(int argc, char **argv)
         return EXIT_FAILURE;
     }
 
+    int status = std::system(("sudo hostnamectl set-hostname " + newHostname).c_str());
```
Write comments about what is going on here

IhorBerehovskyi replied:
Done

Sage-Cat commented on server/src/main.cpp:
```diff
@@ -53,27 +52,25 @@ int main(int argc, char **argv)
 {
     SpdlogConfig::init<SpdlogConfig::LogLevel::Trace>();
 
+    const std::string newHostname = "co2measure.local";
```
"new"?
Why "new"?

IhorBerehovskyi replied:
Done

Sage-Cat commented on client/app/src/main/java/com/myproj/client/fragments/ChartFragment.kt:
```diff
@@ -109,44 +104,24 @@ class ChartFragment : Fragment() {
-            override fun onFailure(call: Call<ResponseBody>, t: Throwable) {
-                // Обробка помилки
+            override fun onFailure(t: Throwable) {
+                // Handle failure
```
At least log it or make a popup for user to see (message box with error)

IhorBerehovskyi replied:
Done

IhorBerehovskyi committed:
General improvements
-Replaced std::cerr by SPDLOG
-Added log at chartFragment
-Renamed field and added comments at main
